ts: TestServerQueryCombinedBatchReducesKVRPCs failed

ts.TestServerQueryCombinedBatchReducesKVRPCs [failed](https://mesolite.cluster.engflow.com/invocations/default/3f177180-7d0b-4a55-8827-596b78e0734e?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvdHM6dHNfdGVzdA==) on master @ [2290b429ab8625d3fc2d2f7adb544ba8fcab6355](https://github.com/cockroachdb/cockroach/commits/2290b429ab8625d3fc2d2f7adb544ba8fcab6355):


```
=== RUN   TestServerQueryCombinedBatchReducesKVRPCs
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestServerQueryCombinedBatchReducesKVRPCs1812207672
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:637: DRPC is enabled (via testserver factory defaults)
    server_test.go:352: individual requests: 6 KV batches, combined request: 1 KV batches
    server_test.go:358: 
        	Error Trace:	pkg/ts/server_test.go:358
        	Error:      	Not equal: 
        	            	expected: 5
        	            	actual  : 6
        	Test:       	TestServerQueryCombinedBatchReducesKVRPCs
        	Messages:   	each individual query should produce one KV BatchRequest
    panic.go:694: -- test log scope end --
test logs left over in: outputs.zip/logTestServerQueryCombinedBatchReducesKVRPCs1812207672
--- FAIL: TestServerQueryCombinedBatchReducesKVRPCs (8.00s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestServerQueryCombinedBatchReducesKVRPCs.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-63634

ts.TestServerQueryCombinedBatchReducesKVRPCs [failed](https://mesolite.cluster.engflow.com/invocations/default/26210825-72ac-4a2b-a238-0be7db7464c3?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvdHM6dHNfdGVzdA==) on master @ [b73593d0150a855f82ee398076748436dabee403](https://github.com/cockroachdb/cockroach/commits/b73593d0150a855f82ee398076748436dabee403):


```
=== RUN   TestServerQueryCombinedBatchReducesKVRPCs
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestServerQueryCombinedBatchReducesKVRPCs96580729
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:637: DRPC is enabled (via testserver factory defaults)
    server_test.go:353: individual requests: 6 KV batches, combined request: 1 KV batches
    server_test.go:359: 
        	Error Trace:	pkg/ts/server_test.go:359
        	Error:      	Not equal: 
        	            	expected: 5
        	            	actual  : 6
        	Test:       	TestServerQueryCombinedBatchReducesKVRPCs
        	Messages:   	each individual query should produce one KV BatchRequest
    panic.go:694: -- test log scope end --
test logs left over in: outputs.zip/logTestServerQueryCombinedBatchReducesKVRPCs96580729
--- FAIL: TestServerQueryCombinedBatchReducesKVRPCs (17.45s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestServerQueryCombinedBatchReducesKVRPCs.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ts/server_test.go
+++ b/pkg/ts/server_test.go
@@ -265,6 +265,17 @@ func TestServerQueryCombinedBatchReducesKVRPCs(t *testing.T) {
 	defer log.Scope(t).Close(t)
 
 	var tsBatchCount atomic.Int64
+	// isTSRead reports whether the request reads from the timeseries keyspace.
+	// The background ts poller writes metrics via Merge requests on the same
+	// keyspace; filtering by request type prevents a poll inflating the count.
+	isTSRead := func(ru kvpb.RequestUnion) bool {
+		inner := ru.GetInner()
+		switch inner.(type) {
+		case *kvpb.GetRequest, *kvpb.ScanRequest:
+			return bytes.HasPrefix(inner.Header().Key, keys.TimeseriesPrefix)
+		}
+		return false
+	}
 	s := serverutils.StartServerOnly(t, base.TestServerArgs{
 		DefaultTestTenant: base.TestIsSpecificToStorageLayerAndNeedsASystemTenant,
 		Knobs: base.TestingKnobs{
@@ -274,10 +285,7 @@ func TestServerQueryCombinedBatchReducesKVRPCs(t *testing.T) {
 					_ context.Context, req *kvpb.BatchRequest,
 				) *kvpb.Error {
 					for _, ru := range req.Requests {
-						if bytes.HasPrefix(
-							ru.GetInner().Header().Key,
-							keys.TimeseriesPrefix,
-						) {
+						if isTSRead(ru) {
 							tsBatchCount.Add(1)
 							break
 						}

